### DISCUSSION TITLE
Support for Tasmota-Zigbee: TS0601 Tuya Smart Zigbee Soil Tester Temperature and Humidity Meter

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#22178] [Ideas]
Can this sensor be included in Tasmota-Zigbee?

![image](https://github.com/user-attachments/assets/7855453b-215e-4ad0-abe0-e50f68edbb7f)

https://nl.aliexpress.com/item/1005007329498149.html

Device Info:
{"Device":"0xE168","Name":"SoilSensorOutside","IEEEAddr":"0xA4C13892D835467F","ModelId":"TS0601","Manufacturer":"_TZE284_aao3yzhs","Endpoints":[1],"Config":["~01.1"],"Reachable":true,"LastSeen":56,"LastSeenEpoch":1726914898,"LinkQuality":13}}

Percentage of moilsture in soil (EF00/0203): 
{"ZbReceived":{"0xE168":{"Device":"0xE168","Name":"SoilSensorOutside","EF00/0203":100,"Endpoint":1,"LinkQuality":16}}}

Battery level (0000/FFE2): 
{"ZbReceived":{"0xE168":{"Device":"0xE168","Name":"SoilSensorOutside","AppVersion":77,"0000/FFE2":56,"0000/FFE4":0,"Endpoint":1,"LinkQuality":3}}}

Temperature (EF00/0205): 
{"ZbReceived":{"0x692D":{"Device":"0x692D","Name":"SoilSensor01","EF00/0205":217,"Endpoint":1,"LinkQuality":8}}}

Humidity? (EF00/020F): {"ZbReceived":{"0x692D":{"Device":"0x692D","Name":"SoilSensor01","EF00/020F":49,"Endpoint":1,"LinkQuality":29}}}

**Comment by s-hadinger** [comment]
Yes. It should be easy to map the Tuya attributes to standard zigbee attributes. 

And you can also publish the sensor values via Matter

**Comment by sisamiwe** [comment]
@jeroenst

Have you been successful?
I would like to connect this sensor also to Tasmota via Z2T

**Reply by jeroenst** [reply]
Yes, but I interpret the values with nodered.

**Reply by quarkcz** [reply]
Can you please help me interpret the values? I don't understand your reply "with nodered".

I have this log from the sensor and have no idea, how to get temperature/moisture/baterry values :-(

```
SENSOR = {"ZbReceived":{"0xB161":{"Device":"0xB161","EF00/0205":254,"Endpoint":1,"LinkQuality":145}}}
SENSOR = {"ZbReceived":{"0xB161":{"Device":"0xB161","Cluster":10,"Read":[7],"ReadNames":{"LocalTime":true},"Endpoint":1,"LinkQuality":116}}}
SENSOR = {"ZbReceived":{"0xB161":{"Device":"0xB161","0000/FFCF":14400,"Endpoint":1,"LinkQuality":120}}}
SENSOR = {"ZbReceived":{"0xB161":{"Device":"0xB161","AppVersion":77,"0000/FFE2":56,"0000/FFE4":1,"Endpoint":1,"LinkQuality":134}}}
SENSOR = {"ZbReceived":{"0xB161":{"Device":"0xB161","EF00?11":"000476","Endpoint":1,"LinkQuality":134}}}
SENSOR = {"ZbReceived":{"0xB161":{"Device":"0xB161","EF00/0203":0,"Endpoint":1,"LinkQuality":134}}}
```

My sensor box say model "HZ-SL04"

**Reply by joba-1** [reply]
I read this yesterday: https://tasmota.github.io/docs/Zigbee/#definition-file

Seems very relevant to your problem.

As I understand it the hex values appear because tuja uses non standard values to identify attributes like temperature - you have to find out what means what and how to convert into correct units and write that in a zb file.

**Reply by sfromis** [reply]
Here's a Tuya-related definition file:
https://github.com/arendst/Tasmota/blob/development/tasmota/zigbee/legacy_tuya.zb

It has a parsing for `EF00/0203` being temperature, which likely is not relevant to your device, when Tuya creates such a bad mess, but it does illustrate how those attributes can be named.

**Reply by quarkcz** [reply]
Based on first post, `EF00/0203` is moisture and `EF00/0205` temperature, but I would like to be sure, `0000/FFE2` is a battery (I would expect more than 56% for my batteries) and what's the rest?

```
"EF00/0205":254       #Temperature based on first comment
"0000/FFCF":14400
"EF00/0203":0         #Percentage of moilsture in soil based on first comment
"EF00?11":"000476"
"0000/FFE2":56        #Battery level based on first comment
"0000/FFE4":1
```

**Reply by s-hadinger** [reply]
That’s the problem with Tuya. Values depend on device’s

**Reply by chrisp1001** [reply]
0000/FFE2 is also showing 56 with my sensor, so looks like it is something static. How would you interpret EF00/0205? What kind of unit should it be?

**Reply by sfromis** [reply]
Read the first post here in this thread. With `EF00/0205` very probably being temperature, the value `"EF00/0205":217` looks a lot like degrees Celcius with 1 decimal. That post also mentions `0000/FFE2` being battery level, which hopefully would not be jumping around.
